api/src/test/java/org/openmrs/module/locationbasedaccess/aop/UserSearchAdviserTest.java

```diff
@@ -0,0 +1,460 @@
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        assertEquals(normalUser.getUserProperties().size(), 0);
+        
+        List<User> userList = userService.getUsersByName("User","accessLocation",false);
+        assertEquals(0, userList.size());
+    }
+    
+    @Test
+    public void getUserByName_nonAdminUserWithLocationPropertyShouldGetOnlyAccessibleUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER2_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        Location location = locationService.getLocation(DEMO_LOCATION_ID);
+
+        assertEquals(normalUser.getUserProperties().size(), 1);
+        String locationUserProperty = normalUser.getUserProperty("locationUuid");
+        assertNotNull(locationUserProperty);
+        assertEquals(locationUserProperty, location.getUuid());
+        
+        List<User> userList = userService.getUsersByName("User","accessLocation",false);
+        assertEquals(userList.size(), 1);
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),location.getUuid());
+    }
+    
+    @Test
+    public void getUserByPerson_adminCanAccessUsersFromAllLocations(){
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location = locationService.getLocation(DEMO_LOCATION_ID);
+        
+        Person person1 = personService.getPerson(DEMO_USER1_ID);
+        List<User> userList1 = userService.getUsersByPerson(person1,false);
+        assertEquals(userList1.size(), 1);
+        assertNotNull(userList1.get(0));
+        assertEquals(userList1.get(0).getUserProperty(DEMO_USER_PROPERTY),"");
+        
+        Person person2 = personService.getPerson(DEMO_USER2_ID);
+        List<User> userList2 = userService.getUsersByPerson(person2,false);
+        assertEquals(userList2.size(), 1);
+        assertNotNull(userList2.get(0));
+        assertEquals(userList2.get(0).getUserProperty(DEMO_USER_PROPERTY),location.getUuid());
+    }
+    
+    @Test
+    public void getUserByPerson_nonAdminUserWithOutLocationPropertyShouldNotGetUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        assertEquals(normalUser.getUserProperties().size(), 0);
+        
+        Person person1 = personService.getPerson(DEMO_USER1_ID);
+        List<User> userList1 = userService.getUsersByPerson(person1,false);
+        assertEquals(0, userList1.size());

```

Comment:
Here is the place that the error occurred. 
Actually, you are trying to get the same user through the authenticated user (Both are DEMO_USER1_ID). It's allowed in this use case since the user can get details about his object :smile: . So the `userList1.size()` should contains that authenticated user (size = 1)

You need to check for, Can a user get another user details without location user property? 
Here DEMO_USER1_ID can't access DEMO_USER2_ID. So `userList2.size()` will be empty (size = 0)

Reply:
Can you change your code to getPerson from personId, not the UserId. 
Both ID can be the same, but you should represent those using different names here. Otherwise, it may hard to understand the concepts.

Reply:
Did it @suthagar23  😄